packages/client/src/app/app.component.ts

```diff
@@ -21,42 +21,49 @@ import { NotificationBase } from '@pulp-fiction/models/notifications';
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.less']
 })
-export class AppComponent implements OnInit, AfterViewInit {
+export class AppComponent implements OnInit, AfterViewInit, OnDestroy {
   @ViewChild('sidenav', {static: true}) sidenav: ElementRef;
 
+  @Select(UserState.currUser) currentUser$: Observable<FrontendUser>;
+  currentUserSubscription: Subscription;
+  currentUser: FrontendUser;
+
   sidenavOpened: boolean;
 
   title = 'offprint';
-  currentUser: FrontendUser;
 
   loading = false;
   loadingLogin = false;
   footerStats: FrontPageStats;
   rotatingSlogan: string;
 
   notifications: NotificationBase[];
+  numNotifs: number = 0;
 
-  constructor(private router: Router, private authService: AuthService, private statsService: StatsService,
+  constructor(private router: Router, private statsService: StatsService,
     private nagBarService: NagBarService, public loader: LoadingBarService, private notif: NotificationsService) {
-    this.authService.currUser.subscribe(x => {
-      this.currentUser = x;
-    });
 
     this.fetchFrontPageStats();
 
-    interval(300000).pipe(flatMap(() => this.notif.getUnreadNotifications())).subscribe(data => {
-      this.notifications = data;
+    this.currentUserSubscription = this.currentUser$.subscribe(x => {
+      this.currentUser = x;
     });
-
-    // Sets the current site theme based on user preference
+    
     if (this.currentUser) {
-        this.changeTheme(PredefinedThemes[this.currentUser.profile.themePref]);     
+        // Sets the current site theme based on user preference
+        this.changeTheme(PredefinedThemes[this.currentUser.profile.themePref]);   
+
+        // Starts fetching notifications updates from the server
+        interval(300000).pipe(flatMap(() => this.notif.getUnreadNotifications())).subscribe(data => {
```

Comment:
Probably better practice to stick this in a `FIVE_MINUTES` const or something--magic numbers bad.

Reply:
true